kademlia/storage.py

```diff
@@ -7,10 +7,10 @@
 class IStorage:
     """
     Local storage for this node.
-    IStorage implementations must store and return values as a key value pair.
-    IStorage special behavior (ie: appending to a list instead of setting) must
-    work in only special cases as the default case (key,value pair) is required
-    for the DHT's functioning.
+    IStorage implementations of get must return the same type as put in by set
+    For example: if and IStorage implementation appends to list instead of
+    replacing the value, it must return one of those list values and not the
+    whole list.
```

Comment:
The `IStorage` interface docs do make it clear that `__setitem__` is used to "Set a key to the given value."  To append to a list for a key would result in a lack of conformity to the interface.  The Kademlia node expects to receive a single key/value in a `STORE` call, and upon receipt of a `FIND_VALUE` must return the same (single) value it most recently received from a `STORE`.  This behavior is defined in section 2.3 of the [paper](https://pdos.csail.mit.edu/~petar/papers/maymounkov-kademlia-lncs.pdf).